src/palette/internal/palettelayout.cpp

```diff
@@ -857,8 +1035,383 @@ void PaletteLayout::layout(TimeSig* item, const Context& ctx)
     item->setDrawArgs(drawArgs);
 }
 
-//! TODO
-void PaletteLayout::layout(Volta*, const Context&)
+void PaletteLayout::layout(Vibrato* item, const Context& ctx)
+{
+    layoutLine(static_cast<SLine*>(item), ctx);
+}
+
+void PaletteLayout::layout(VibratoSegment* item, const Context&)
+{
+    if (item->spanner()->placeBelow()) {
+        item->setPosY(item->staff() ? item->staff()->height() : 0.0);
```

Comment:
It seems that we don't need this item->staff() anymore, so we can rewrite this line as follows:

`item->setPosY(0.0);`